update-go-version.sh

```diff
@@ -12,6 +12,8 @@ echo "Updating golang version to $latest_version"
 
 go mod edit -go ${golang_base_version}
 sed -i "s,^FROM registry.svc.ci.openshift.org/openshift/release:golang-1\... AS builder\$,FROM registry.svc.ci.openshift.org/openshift/release:golang-${golang_base_version} AS builder," images/openshift-ci/Dockerfile
```

Comment:
@cfergeau We should also update this using `registry.ci.openshift` instead `registry.svc.ci.openshift` .